protocols/kad/src/kbucket.rs

```diff
@@ -150,17 +187,14 @@ where
     TVal: Clone,
 {
     /// Creates a new, empty Kademlia routing table with entries partitioned
-    /// into buckets as per the Kademlia protocol.
-    ///
-    /// The given `pending_timeout` specifies the duration after creation of
-    /// a [`PendingEntry`] after which it becomes eligible for insertion into
-    /// a full bucket, replacing the least-recently (dis)connected node.
-    pub(crate) fn new(local_key: TKey, pending_timeout: Duration) -> Self {
+    /// into buckets as per the Kademlia protocol using the provided config.
+    pub(crate) fn new(local_key: TKey, config: KBucketConfig) -> Self {
         KBucketsTable {
             local_key,
             buckets: (0..NUM_BUCKETS)
-                .map(|_| KBucket::new(pending_timeout))
+                .map(|_| KBucket::new(config.clone()))
                 .collect(),
+            bucket_size: config.bucket_size,
             applied_pending: VecDeque::new(),
         }
```

Comment:
Although its requires nor does it really cost much, but we could skip cloning by doing `let bucket_size = config.bucket_size` and passing that to `KBucketsTable`.

Reply:
IIUC because of the `map` we are forced to clone or copy anyway. I added the `Copy` trait to `KBucketConfig`. Let me know if you think there is a better way to proceed.